src/qaqc/download_and_flag_fxns/add_biofilm_flag.R

```diff
@@ -21,37 +28,70 @@ add_biofilm_flag <- function(df){
 
       } else {
 
+        # If not enough data, set slope to 0
         no_slope <- 0
 
         return(no_slope)
 
       }
     }
 
-    drift_tested <- df %>%
-      data.table::data.table() %>%
-      #bind_rows(flagged_data_dfs[["prospect-Turbidity"]]) %>%
-      #filter(as_date(DT_round) >= "2022-10-11" & as_date(DT_round) <= "2022-10-15") %>%
-      #mutate(r_squared1 = data.table::frollapply(mean, n = 96, FUN = progressive_drift, align = "left", fill = NA)) %>%
-      mutate(r_squared2 = data.table::frollapply(mean, n = 96, FUN = progressive_drift, align = "right", fill = NA)) %>%
-      #mutate(r_squared3 = data.table::frollapply(mean, n = 96, FUN = progressive_drift, align = "center", fill = NA)) %>%
-      #mutate(r_squared4 = data.table::frollapply(mean, n = 288, FUN = progressive_drift, align = "left", fill = NA)) %>%
-      mutate(r_squared5 = data.table::frollapply(mean, n = 288, FUN = progressive_drift, align = "right", fill = NA)) %>%
-      #mutate(r_squared6 = data.table::frollapply(mean, n = 288, FUN = progressive_drift, align = "center", fill = NA)) %>%
-      # r-squared can only be in the positive direction for biofilm drift:
-      add_flag((#r_squared1 >= 0.7 |
-                  r_squared2 >= 0.7 |
-                  #r_squared3 >= 0.7 |
-                #r_squared4 >= 0.8 |
-                  r_squared5 >= 0.8 #|
-                  #r_squared6 >= 0.8
-                )  & !grepl("biofilm", flag), "biofilm") #%>%
-      #select(-c(r_squared1, r_squared2, r_squared3)
-
-    return(drift_tested)
-
-  } else { return(df) }
-  }
+    # Function to check if a selected time window's mean R-squared value is
+    # at least 60% (i.e., fairly high linear relationship with time indicates a "steady" drift)
+    check_too_steady <- function(x) {
+      mean(x) >= 0.60
+    }
```

Comment:
Suggestion: I wonder if we would want to statistically verify this R2 value.

Question: Is this checking the R2 value for the 3 hour period that we pulled in?

Reply:
It grabs the average R2 value across both a moving window of 24-hour data and 3-day data . Can definitely be improved!